Build Failure: jenkins: io.pravega.segmentstore.server.containers.StreamSegmentMapperTests.testGetOrAssignStreamSegmentIdWithMetadataLimit

**Problem description**
Observing the following build failure on jenkins.
Error Message

java.lang.AssertionError: Unexpected outcome when trying to map a segment name to a full metadata that cannot be cleaned. Exception thrown was of unexpected type: io.pravega.segmentstore.contracts.TooManyActiveSegmentsException: [Container 2] The maximum number of active Segments (0) has been reached..

Stacktrace

```
java.lang.AssertionError: Unexpected outcome when trying to map a segment name to a full metadata that cannot be cleaned. Exception thrown was of unexpected type: io.pravega.segmentstore.contracts.TooManyActiveSegmentsException: [Container 2] The maximum number of active Segments (0) has been reached..
	at org.junit.Assert.fail(Assert.java:88)
	at io.pravega.test.common.AssertExtensions.assertThrows(AssertExtensions.java:86)
	at io.pravega.segmentstore.server.containers.StreamSegmentMapperTests.testGetOrAssignStreamSegmentIdWithMetadataLimit(StreamSegmentMapperTests.java:375)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
```

**Problem location**

StreamSegmentMapperTests

--- a/segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentMapperTests.java
+++ b/segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentMapperTests.java
@@ -80,7 +80,7 @@ public class StreamSegmentMapperTests extends ThreadPooledTestSuite {
 
     @Override
     protected int getThreadPoolSize() {
-        return 5;
+        return 1;
     }
 
     /**
@@ -363,9 +363,9 @@ public class StreamSegmentMapperTests extends ThreadPooledTestSuite {
         };
         val mapper1 = new StreamSegmentMapper(context.metadata, context.operationLog, context.stateStore, noOpCleanup, context.storage, executorService());
         AssertExtensions.assertThrows(
-                "Unexpected outcome when trying to map a segment name to a full metadata that cannot be cleaned.",
+                "Unexpected outcome when trying to map a segment to a full metadata that cannot be cleaned.",
                 () -> mapper1.getOrAssignStreamSegmentId(segmentName, TIMEOUT),
-                ex -> ex instanceof TooManyActiveSegmentsException && ((TooManyActiveSegmentsException) ex).getContainerId() == exceptionCounter.get());
+                ex -> ex instanceof TooManyActiveSegmentsException);
         Assert.assertEquals("Unexpected number of attempts to map.", 2, exceptionCounter.get());
         Assert.assertTrue("Cleanup was not invoked.", cleanupInvoked.get());
 
@@ -373,9 +373,9 @@ public class StreamSegmentMapperTests extends ThreadPooledTestSuite {
         exceptionCounter.set(0);
         cleanupInvoked.set(false);
         AssertExtensions.assertThrows(
-                "Unexpected outcome when trying to map a segment name to a full metadata that cannot be cleaned.",
+                "Unexpected outcome when trying to map a transaction to a full metadata that cannot be cleaned.",
                 () -> mapper1.getOrAssignStreamSegmentId(transactionName, TIMEOUT),
-                ex -> ex instanceof TooManyActiveSegmentsException && ((TooManyActiveSegmentsException) ex).getContainerId() == exceptionCounter.get());
+                ex -> ex instanceof TooManyActiveSegmentsException);
         Assert.assertEquals("Unexpected number of attempts to map.", 2, exceptionCounter.get());
         Assert.assertTrue("Cleanup was not invoked.", cleanupInvoked.get());
 

--- a/segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentMapperTests.java
+++ b/segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/StreamSegmentMapperTests.java
@@ -80,7 +80,7 @@ public class StreamSegmentMapperTests extends ThreadPooledTestSuite {
 
     @Override
     protected int getThreadPoolSize() {
-        return 1;
+        return 5;
     }
 
     /**

